Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/shortcuts/AppShortcutsHelper.kt

```diff
@@ -1,64 +1,96 @@
 package de.rki.coronawarnapp.util.shortcuts
 
+import android.Manifest
 import android.content.Context
 import android.content.Intent
+import androidx.core.content.PermissionChecker
+import androidx.core.content.PermissionChecker.PERMISSION_GRANTED
 import androidx.core.content.pm.ShortcutInfoCompat
 import androidx.core.content.pm.ShortcutManagerCompat
 import androidx.core.graphics.drawable.IconCompat
 import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.coronatest.CoronaTestRepository
+import de.rki.coronawarnapp.storage.OnboardingSettings
 import de.rki.coronawarnapp.ui.launcher.LauncherActivity
 import de.rki.coronawarnapp.util.AppShortcuts
 import de.rki.coronawarnapp.util.di.AppContext
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.withContext
+import timber.log.Timber
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class AppShortcutsHelper @Inject constructor(@AppContext private val context: Context) {
-
+class AppShortcutsHelper @Inject constructor(
+    @AppContext private val context: Context,
+    private val onboardingSettings: OnboardingSettings,
+    private val coronaTestRepository: CoronaTestRepository
+) {
     suspend fun restoreAppShortcut() = withContext(Dispatchers.IO) {
+        // No shortcuts if test result is positive
+        if (coronaTestRepository.allCoronaTests.first().any { it.isPositive }) {
+            Timber.i("[AppShortcuts] Remove all shortcut items since exposure submission result is positive")
+            removeAppShortcuts()
+            return@withContext
+        }
+
+        // No shortcuts if not onboarded
+        if (!onboardingSettings.isOnboarded) {
```

Comment:
Sure, shortcuts should be available after overall onboarding , but we need to check every section on-boarding too.
For these screens there is an onboarding screen `CheckIns` and  `Certificates`

Reply:
I tried to add these checks in https://github.com/corona-warn-app/cwa-app-android/pull/4854/commits/91283f04a599aaa23c5cf9838eb1fb322e0fd712